posthog/models/user.py

```diff
@@ -53,7 +53,49 @@ class UserManager(BaseUserManager):
     """Define a model manager for User model with no username field."""
 
     def get_queryset(self):
-        return super().get_queryset().defer(*DEFERED_ATTRS)
+        # Optimize user queries by preloading essential fields and related organization data.
```

Comment:
@benjackwhite FYI I needed to prefetch fields here to ensure there were no extra queries are being made